### PR TITLE
Add package `pycryptodome`

### PR DESCRIPTION
**Pull request by Starz0r**
### Description

Add cryptography package `pycryptodome`

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by Starz0r** [comment]
I'm unsure what the lint tests want, I've removed and added the newline, but it fails either way?

Perhaps it's just flaky 😅.

**Comment by hoodmane** [comment]
I don't think the lint check is flakey. The precommit ci job says `push permission was denied -- perhaps the "Allow edits by maintainers" box was unchecked?` If you make sure that box is checked the CI job should be able to fix the lints automatically.

**Comment by Starz0r** [comment]
>  perhaps the "Allow edits by maintainers" box was unchecked?` If you make sure that box is checked the CI job should be able to fix the lints automatically.

Did not know that was mandatory to ensure those tests pass!

**Comment by Starz0r** [comment]
The test I wrote also fails, despite being 90% similar to the one the actually passes: https://github.com/pyodide/pyodide/pull/2966.

It looks like the native C library fails to load for some reason, but I'm unsure why since both packages utilize them, why does this one fail to load it? Going to need some insight from someone else who knows more about this.

**Comment by hoodmane** [comment]
Thanks @Starz0r! One of us will try to look into it soon.

File: packages/pycryptodome/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package:
+  name: pycryptodome
+  version: 3.15.0
+source:
+  url: https://files.pythonhosted.org/packages/11/e4/a8e8056a59c39f8c9ddd11d3bc3e1a67493abe746df727e531f66ecede9e/pycryptodome-3.15.0.tar.gz
+  sha256: 9135dddad504592bcc18b0d2d95ce86c3a5ea87ec6447ef25cfedea12d6018b8
+test:
```
Comment (hoodmane) [inline packages/pycryptodome/meta.yaml:7]: I think the following should fix it:
```suggestion
build:
  exports: "requested"
test:
```

**Comment by hoodmane** [comment]
Strange build error looks unrelated. I'll rerun it and hope it goes away.

**Comment by hoodmane** [comment]
@Starz0r you should merge main here to get #2972

File: packages/pycryptodome/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  name: pycryptodome
+  version: 3.15.0
+source:
+  url: https://files.pythonhosted.org/packages/11/e4/a8e8056a59c39f8c9ddd11d3bc3e1a67493abe746df727e531f66ecede9e/pycryptodome-3.15.0.tar.gz
+  sha256: 9135dddad504592bcc18b0d2d95ce86c3a5ea87ec6447ef25cfedea12d6018b8
+build:
+  exports: "requested"
+test:
+  imports:
+    - pycryptodome
```
Comment (hoodmane) [inline packages/pycryptodome/meta.yaml:11]:
```suggestion
    - Crypto
```

**Comment by hoodmane** [comment]
Thanks @Starz0r!

**Pull request merged**